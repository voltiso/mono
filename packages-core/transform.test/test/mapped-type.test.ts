// â €â“¥ 2022     ðŸŒ©    ðŸŒ©     â €   â €
// â €         ðŸŒ© VÍ›oÍ›Í›Í›ltÍ›Í›Í›iÍ›Í›Í›Í›soÍ›Í›Í›.comâ €  â €â €â €

import * as fs from 'node:fs/promises'

describe('mapped-type', () => {
	it('works', async () => {
		expect.hasAssertions()

		const file = await fs.readFile('./dist/cjs/mapped-type.d.ts')
		expect(file.toString()).toMatchSnapshot()
	})
})
