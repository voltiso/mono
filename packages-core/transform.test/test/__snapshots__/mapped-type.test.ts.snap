// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapped-type works 1`] = `
"/** @inline */
export declare type DeepPartial_<T> = [
    {
        [k in keyof T]?: DeepPartial_<T[k]>;
    }
][0];
export declare type _ObjectLike = {
    OPTIONS: {
        shape: any;
        Output: any;
        Input: any;
    };
};
export declare type _GetDeepPartial_<This> = This extends _ObjectLike ? {
    Output: DeepPartial_<This['OPTIONS']['Output']>;
} : never;
//# sourceMappingURL=mapped-type.d.ts.map"
`;
